modules/__init__.py

```diff
@@ -1,38 +1,26 @@
-from modules.base import Module, noparts, check_solve_cmd, gif_append, gif_output
+from os.path import basename, dirname, join as path_join
+from glob import glob
+import importlib
+import inspect
+
+LIST_OF_VANILLA_MODULES=['Button','ComplicatedWires','Keypad','Maze','Memory','MorseCode','Password','SimonSays','WhosOnFirst','Wires','WireSequence']
 
-from modules.wires import Wires, ComplicatedWires, WireSequence
-from modules.button import Button
-from modules.keypad import Keypad
-from modules.simon import SimonSays
-from modules.whos_on_first import WhosOnFirst, ThirdBase
-from modules.memory import Memory
-from modules.morse import MorseCode
-from modules.maze import Maze
-from modules.password import Password
+VANILLA_MODULES = {}
+MODDED_MODULES = {}
 
-from modules.connection_check import ConnectionCheck
-from modules.hexamaze import Hexamaze
+def register_module(module):
+    module_name=inspect.getmro(module)[0].__name__
+    category = VANILLA_MODULES if module_name in LIST_OF_VANILLA_MODULES else MODDED_MODULES
+    for identifier in module.identifiers:
+        category[identifier] = module
 
-VANILLA_MODULES = {
-    "wires": Wires,
-    "button": Button,
-    "keypad": Keypad,
-    "simonSays": SimonSays,
-    "whosOnFirst": WhosOnFirst,
-    "memory": Memory,
-    "morseCode": MorseCode,
-    "complicatedWires": ComplicatedWires,
-    "wireSequence": WireSequence,
-    "maze": Maze,
-    "password": Password,
-}
+# This has to be here to avoid cyclic imports.
+from modules.base import Module, noparts, check_solve_cmd, gif_append, gif_output
 
-MODDED_MODULES = {
-    "connectionCheck": ConnectionCheck,
-    "hexamaze": Hexamaze,
-    "thirdBase": ThirdBase,
-}
+for module_file in glob(path_join(dirname(__file__), "*.py")):
+    module_name = basename(module_file)[:-3]
+    importlib.import_module('modules.' + module_name)
 
 async def cmd_modules(channel, author, parts):
-    list_ = lambda d: ', '.join(f"`{x}`" for x in d)
+    def list_(d): return ', '.join(f"`{x}`" for x in d)
     await channel.send(f"Available modules:\nVanilla: {list_(VANILLA_MODULES)}\nModded: {list_(MODDED_MODULES)}")
```

Comment:
Don't squash other people's commits into yours, especially if the changes are unrelated.